openverse_catalog/templates/template_provider.py_template

```diff
@@ -0,0 +1,163 @@
+DAGs.md. Update it to include any relevant information that you'd like to
+be documented.
+
+Content Provider:       {provider}
+
+ETL Process:            Use the API to identify all CC licensed images.
+
+Output:                 TSV file containing the images and the
+                        respective meta-data.
+
+Notes:                  {endpoint}
+"""
+import logging
+
+from airflow.models import Variable
+from common import constants
+from common.licenses import get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+logger = logging.getLogger(__name__)
+
+
+class {provider}DataIngester(ProviderDataIngester):
+    """
+    This is a template for a ProviderDataIngester.
+
+    Methods are shown with example implementations. Adjust them to suit your API.
+    """
+
+    # TODO: Add the provider constants to `common.loader.provider_details.py`
+    providers = {
+        {provider_configuration}
+    }
+    endpoint = "{endpoint}"
+    # TODO The following are set to their default values. Remove them if the defaults
+    # are acceptable, or override them.
+    delay = 1
+    retries = 3
+    batch_limit = 100
+    headers = {}
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        # On the first request, `prev_query_params` will be `None`. We can detect this
+        # and return our default params.
+        if not prev_query_params:
+            # TODO: Return your default params here. `batch_limit` is not automatically added to
+            # the params, so make sure to add it here if you need it!
+            # TODO: If you need an API key, add the (empty) key to `openverse_catalog/env.template`
+            # Do not hardcode API keys!
+            return {
+                "limit": self.batch_limit,
+                "cc": 1,
+                "offset": 0,
+                "api_key": Variable.get("API_KEY_{screaming_snake_provider}")
+            }
+        else:
+            # TODO: Update any query params that change on subsequent requests.
+            # Example case shows the offset being incremented by batch limit.
+            return {
+                **prev_query_params,
+                "offset": prev_query_params["offset"] + self.batch_limit,
+            }
+
+    def get_batch_data(self, response_json):
+        # Takes the raw API response from calling `get` on the endpoint, and returns
+        # the list of records to process.
+        # TODO: Update based on your API.
+        if response_json:
+            return response_json.get("results")
+        return None
+
+    def get_media_type(self, record: dict):
+        # For a given record json, return the media type it represents.
+        # TODO: Update based on your API. TIP: May be hard-coded if the provider only
+        # returns records of one type.
```

Comment:
Maybe include an example in this comment of what it could look like if we were just returning the constant?